### DISCUSSION TITLE
[API] Is there API support for following an external account?

### DISCUSSION THREAD

**Discussion by cfreshman** [mastodon/mastodon#20291] [Q&A] [answered]
As far as I can tell, the [follow API](https://docs.joinmastodon.org/methods/accounts/#:~:text=https%3A//mastodon.example/api/v1/accounts/%3Aid/follow) (```/api/v1/accounts/:id/follow```) only works on accounts within the given Mastodon server

In contrast, the [search API](https://docs.joinmastodon.org/methods/search/#:~:text=https%3A//mastodon.example/api/v2/search) (```/api/v2/search```) is able to look up accounts on other servers

Is there some way to follow accounts on other servers programmatically?

**Comment by trwnh** [comment, marked as answer]
You would have to search for their profile URL via the search API (with `resolve` set to true), which should create a local account ID for them in the database. You can see this ID in the response, which should return an account. Then you pass that local account ID into the follow account API as a path parameter.

Alternatively, since v3.4.0 you can search for their Webfinger address with `/api/v1/accounts/lookup` and the `acct` parameter, although this isn't live on the docs site yet -- there is a PR for updating the docs to 4.0

**Reply by Gargron** [reply]
Lookup will not perform a resolve.

**Reply by cfreshman** [reply]
Nvm, I am an idiot, forgot to ```POST``` instead of ```GET``` (I'm ```POST```ing to my backend first). Not proud of how long I spent on that, but it's working now - I'm using it to sync follows from Twitter to Mastodon (https://freshman.dev/follow-sync)

Thank you for your responses
